backend/compact-connect/lambdas/purchases/handlers/privileges.py

```diff
@@ -151,28 +153,32 @@ def post_purchase_privileges(event: dict, context: LambdaContext):  # noqa: ARG0
     provider_id = _get_caller_provider_id_custom_attribute(event)
     user_provider_data = config.data_client.get_provider(compact=compact_name, provider_id=provider_id)
     provider_record = next((record for record in user_provider_data['items'] if record['type'] == 'provider'), None)
-    license_record = _find_best_license(
+    license_record = _find_latest_active_license(
         [record for record in user_provider_data['items'] if record['type'] == 'license']
     )
-    existing_privilege_jurisdictions = [
-        record['jurisdiction'] for record in user_provider_data['items'] if record['type'] == 'privilege'
-    ]
+
     # this should never happen, but we check just in case
     if provider_record is None:
         raise CCNotFoundException('Provider not found')
     if license_record is None:
-        raise CCNotFoundException('License record not found for this user')
+        raise CCInvalidRequestException('No active license found for this user')
 
     license_jurisdiction = license_record['jurisdiction']
     if license_jurisdiction.lower() in selected_jurisdictions_postal_abbreviations:
         raise CCInvalidRequestException(
             f"Selected privilege jurisdiction '{license_jurisdiction}'" f' matches license jurisdiction'
         )
 
-    for privilege_jurisdiction in existing_privilege_jurisdictions:
-        if privilege_jurisdiction.lower() in selected_jurisdictions_postal_abbreviations:
+    existing_privileges = [record for record in user_provider_data['items'] if record['type'] == 'privilege']
+    # a licensee can only purchase an existing privilege for a jurisdiction
+    # if their existing privilege expiration date does not match their license expiration date
+    for privilege in existing_privileges:
+        if (
+            privilege['jurisdiction'].lower() in selected_jurisdictions_postal_abbreviations
+            and privilege['dateOfExpiration'] == license_record['dateOfExpiration']
```

Comment:
Not about the code, but mind giving me the quick tldr on this logic? A bit confused by the constraint.

Reply:
Great, thanks!